packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1493,87 +1509,43 @@ class SliverAppBar extends StatefulWidget {
   ///   * [SliverAppBar.large], for a large top app bar.
   ///   * https://m3.material.io/components/top-app-bar/overview, the Material 3
   ///     app bar specification.
-  factory SliverAppBar.medium({
-    Key? key,
-    Widget? leading,
-    bool automaticallyImplyLeading = true,
-    Widget? title,
-    List<Widget>? actions,
-    Widget? flexibleSpace,
-    PreferredSizeWidget? bottom,
-    double? elevation,
-    double? scrolledUnderElevation,
-    Color? shadowColor,
-    Color? surfaceTintColor,
-    bool forceElevated = false,
-    Color? backgroundColor,
-    Color? foregroundColor,
-    IconThemeData? iconTheme,
-    IconThemeData? actionsIconTheme,
-    bool primary = true,
-    bool? centerTitle,
-    bool excludeHeaderSemantics = false,
-    double? titleSpacing,
-    double? collapsedHeight,
-    double? expandedHeight,
-    bool floating = false,
-    bool pinned = true,
-    bool snap = false,
-    bool stretch = false,
-    double stretchTriggerOffset = 100.0,
-    AsyncCallback? onStretchTrigger,
-    ShapeBorder? shape,
-    double toolbarHeight = _MediumScrollUnderFlexibleConfig.collapsedHeight,
-    double? leadingWidth,
-    TextStyle? toolbarTextStyle,
-    TextStyle? titleTextStyle,
-    SystemUiOverlayStyle? systemOverlayStyle,
-  }) {
-    return SliverAppBar(
-      key: key,
-      leading: leading,
-      automaticallyImplyLeading: automaticallyImplyLeading,
-      flexibleSpace: flexibleSpace ?? _ScrollUnderFlexibleSpace(
-        hasLeading: leading != null,
-        title: title,
-        actions: actions,
-        foregroundColor: foregroundColor,
-        variant: _ScrollUnderFlexibleVariant.medium,
-        centerCollapsedTitle: centerTitle,
-        primary: primary,
-        leadingWidth: leadingWidth,
-        titleSpacing: titleSpacing,
-      ),
-      bottom: bottom,
-      elevation: elevation,
-      scrolledUnderElevation: scrolledUnderElevation,
-      shadowColor: shadowColor,
-      surfaceTintColor: surfaceTintColor,
-      forceElevated: forceElevated,
-      backgroundColor: backgroundColor,
-      foregroundColor: foregroundColor,
-      iconTheme: iconTheme,
-      actionsIconTheme: actionsIconTheme,
-      primary: primary,
-      centerTitle: centerTitle,
-      excludeHeaderSemantics: excludeHeaderSemantics,
-      titleSpacing: titleSpacing,
-      collapsedHeight: collapsedHeight ?? _MediumScrollUnderFlexibleConfig.collapsedHeight,
-      expandedHeight: expandedHeight ?? _MediumScrollUnderFlexibleConfig.expandedHeight,
-      floating: floating,
-      pinned: pinned,
-      snap: snap,
-      stretch: stretch,
-      stretchTriggerOffset: stretchTriggerOffset,
-      onStretchTrigger: onStretchTrigger,
-      shape: shape,
-      toolbarHeight: toolbarHeight,
-      leadingWidth: leadingWidth,
-      toolbarTextStyle: toolbarTextStyle,
-      titleTextStyle: titleTextStyle,
-      systemOverlayStyle: systemOverlayStyle,
-    );
-  }
+  const SliverAppBar.medium({
```

Comment:
Since these are not factories anymore, should some of the assertions from the base constructor be evaluated here as well?

Reply:
Yes! I totally missed that. Great catch.